newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/PolygonsList.js

```diff
@@ -208,23 +208,20 @@ const PolygonsList = (props: PolygonsListProps) => {
     restoreCollisionMask,
   } = props;
 
-  const addCollisionMask = () => {
-    const newPolygon = gd.Polygon2d.createRectangle(32, 32);
+  const addCollisionMask = React.useCallback(() => {
+    const newPolygon = gd.Polygon2d.createRectangle(spriteWidth, spriteHeight);
     newPolygon.move(spriteWidth / 2, spriteHeight / 2);
     polygons.push_back(newPolygon);
     onPolygonsUpdated();
-  };
+  }, [spriteHeight, spriteWidth, polygons, onPolygonsUpdated]);
 
   React.useEffect(
     () => {
       if (polygons.size() === 0) {
-        const newPolygon = gd.Polygon2d.createRectangle(32, 32);
-        newPolygon.move(spriteWidth / 2, spriteHeight / 2);
-        polygons.push_back(newPolygon);
-        onPolygonsUpdated();
+        addCollisionMask();
       }
     },
-    [polygons, spriteWidth, spriteHeight, onPolygonsUpdated]
+    [polygons, spriteWidth, spriteHeight, onPolygonsUpdated, addCollisionMask]
```

Comment:
Nit: You can even remove spriteWidth/spriteHeight/onPolygonsUpdated. 
Per se it's not invalid, just redundant because addCollisionMask will be changed if they are ;)

Reply:
Oh you're right! I'll fix this